src/averageAges.js

```diff
@@ -54,9 +82,32 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+    && hasMother(people, person.mother))
+    : people.filter(person => hasMother(people, person.mother));
+
+  const motherAge = children
+    .map(person => person.born - bornMother(people, person.mother));
+
+  const motherAverageAge = motherAge
+    .reduce((firstWoman, secondWoman) =>
+      (firstWoman + secondWoman)) / motherAge.length;
+
+  return motherAverageAge;
+};
+
+function hasMother(people, motherName) {
+  return people.some(person => person.name === motherName);
+};
+
+function bornMother(people, motherName) {
+  const mother = people.find(person => person.name === motherName);
+
+  return mother.born;
+};
```

Comment:
u can make it much more simpler.
just find all kids using filter and then by reduce count all age differences between kids and mothers